spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -247,11 +247,8 @@ class Analyzer(tableUtils: TableUtils,
            |""".stripMargin)
     }
 
-    val aggMetadata = if (groupByConf.aggregations != null) {
```

Comment:
@pengyu-hou @SophieYu41 Do you know why we add this line here? Currently we are not exporting the derived groupBy feature due to this condition.

Reply:
This looks like a bug to me. Your change makes more sense. 

Can you add some test cases:
- groupBy without aggregation or derivation
- groupBy with aggregation but w/o derivation
- groupBy with aggregation and derivation
- groupBy without aggregation but with derivation